civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -777,6 +778,45 @@ function get_civihr_absence_statuses($status_id = NULL) {
     return $status_id_value;
 }
 
+/**
+ * This function returns the sickness reason label for the sickness reason
+ * option value passed to it.
+ * The sickness reason option group is cached to avoid hitting the Db on
+ * every call.
+ *
+ * @param int $sickness_reason_value
+ *
+ * @return string
+ */
+function get_civihr_absence_sickness_reason_label($sickness_reason_value) {
+    $sicknessReasons = &drupal_static(__FUNCTION__);
+
+    if (!isset($sicknessReasons)) {
+        if ($cache = cache_get('civihr_absence_sickness_reasons')) {
+            $sicknessReasons = $cache->data;
+        }
+        else {
+            try {
+                // Civi init
+                civicrm_initialize();
+                $sicknessReasons = CRM_Core_OptionGroup::values('hrleaveandabsences_sickness_reason', FALSE, FALSE, FALSE, NULL, 'label');
+
+                watchdog('DB hit absence sickness reasons', 'DB');
+            }
+
+            catch (CiviCRM_API3_Exception $e) {
+                $error = $e->getMessage();
+            }
+
+            // Cache the absence sickness reasons for 5 minutes
+            cache_set('civihr_absence_sickness_reasons', $sicknessReasons, 'cache', time() + 360);
```

Comment:
360 is a magic number, always try to avoid magic numbers in code, it should be defined as constant right before calling cache_set() 

```php
 const secondsInFiveMinutes = 300; 
 cache_set('civihr_absence_sickness_reasons', $sicknessReasons, 'cache', time() + secondsInFiveMinutes);
```

also hence 5 minutes are 300, not 360 seconds.

also since 'civihr_absence_sickness_reasons' cache key is used twice, and because magic strings are not usually that different from magic numbers, you can also define it as constant at the top of the function and use it instead.

```php
const sicknessReasonsCacheKey = 'civihr_absence_sickness_reasons';
```